cupy/linalg/einsum.py

```diff
@@ -18,22 +18,33 @@ def calc_single_view(ioperand, subscript):
             more than once, calculate diagonal for those axes.
     """
 
-    assert ioperand.ndim == len(subscript)
+    if '@' not in subscript:
```

Comment:
`assert ioperand.ndim == len(subscript) - subscript.count('@')`

Reply:
Because `@` and `...` can contain 0 or more dimension, `ioperand.ndim == len(subscript) - subscript.count('@')` is not always `True`.
So I added assertion l.21- l.24.